DSharpPlus.Caching/Abstractions/AbstractCacheEntry.cs

```diff
@@ -0,0 +1,18 @@
+namespace DSharpPlus.Caching.Abstractions
+{
+    /// <summary>
+    /// Represents an extendable class for more complex cache entries supported by <see cref="ICacheService"/>.
+    /// </summary>
+    public record BaseCacheEntry
+    {
+        /// <summary>
+        /// The key used to store this object into cache.
+        /// </summary>
+        public object Key { get; set; } = null!;
```

Comment:
I feel like for consistency this should be a string. Depending on how entities are cached (e.g. `typeof` keys), this becomes incompatible for external caching solutions (I shall again point to Remora in these dire times.)

Reply:
implemented